src/tcgwars/logic/impl/gen3/UnseenForces.groovy

```diff
@@ -1206,11 +1206,9 @@ public enum UnseenForces implements LogicCardInfo {
           onAttack {
             damage 20
             afterDamage {
-              if (opp.hand) {
-                def list = opp.hand.shuffledCopy().showToMe("Opponent's hand").filterByType(TRAINER)
-                if(list){
-                  list.select("Select a Trainer card to discard").discard()
-                }
+              if (opp.hand && opp.hand.filterByType(TRAINER)) {
```

Comment:
```suggestion
              if (opp.hand) {
```
You can't know if the opponent has a trainer in their hand unless you look at their hand.